Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -1010,7 +1041,8 @@ class CombinatorialKalmanFilter {
                           result_type& result) const {
       const auto& logger = state.options.logger;
       // The tip of the track being smoothed
-      const auto& currentTip = result.trackTips.at(result.iSmoothed);
+      const auto& currentTip =
```

Comment:
Nice. One more thing, since we already have knowledge of the last measurement state, could you simplify the following code (which is used to find the last measurement state) a bit? We could also do this simplfication to the `finalize` function in the Kalman Filter.

Reply:
I added some changes let me know what you think.